include/seqan3/io/sequence_file/format_fasta.hpp

```diff
@@ -329,47 +326,31 @@ class format_fasta
         if (options.fasta_blank_before_id)
             stream_it = ' ';
 
-        std::ranges::copy(id, stream_it);
+        stream_it.write_range(id);
 
         detail::write_eol(stream_it, options.add_carriage_return);
     }
 
     //!\brief Implementation of writing the sequence.
-    template <typename stream_it_t,
-              typename seq_type>
-    void write_seq(stream_it_t & stream_it,
-                   sequence_file_output_options const & options,
-                   seq_type && seq)
+    template <typename stream_it_t, typename seq_type>
+    void write_seq(stream_it_t & stream_it, sequence_file_output_options const & options, seq_type && seq)
     {
+        auto char_sequence = seq | views::to_char;
+
         if (options.fasta_letters_per_line > 0)
         {
-        #if SEQAN3_WORKAROUND_VIEW_PERFORMANCE
-            size_t count = 0;
-            for (auto c : seq)
+            auto it = std::ranges::begin(char_sequence);
+
+            while (it != std::ranges::end(char_sequence))
             {
-                stream_it = to_char(c);
-                if (++count % options.fasta_letters_per_line == 0)
-                    detail::write_eol(stream_it, options.add_carriage_return);
-            }
-            if (count % options.fasta_letters_per_line != 0)
+                auto end_it = std::ranges::next(it, options.fasta_letters_per_line, std::ranges::end(char_sequence));
+                it = stream_it.write_range(std::ranges::subrange<decltype(it), decltype(end_it)>{it, end_it});
                 detail::write_eol(stream_it, options.add_carriage_return);
```

Comment:
Can you add a note inside the code or as part of the functions documentation that 
a) The performance might be suboptimal for non-random-access-ranges 
b) `views::interleave` is probably the way to go but that needs performance-tuning itself

Reply:
I still think adding a note somewhere about views::interleave wouldn't be too bad. Since that was created exactly for this purpose and actually preserves size and random-access...

Reply:
I did add this comment :) but further up before the whole while loop